src/js/select2/defaults.js

```diff
@@ -318,6 +318,8 @@ define([
       amdBase: 'select2/',
       amdLanguageBase: 'select2/i18n/',
       closeOnSelect: true,
+      dropdownAutoWidth: false,
+      dropdownMaxWidth: '100%',
```

Comment:
If you can get away with using `css.maxWidth` below, then you should be able to just not include this in the defaults.